zeebe/atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/MessagingMetricsImpl.java

```diff
@@ -7,97 +7,163 @@
  */
 package io.atomix.cluster.messaging.impl;
 
+import static io.atomix.cluster.messaging.impl.MessagingMetricsDoc.*;
+
 import io.camunda.zeebe.util.CloseableSilently;
-import io.prometheus.client.Counter;
-import io.prometheus.client.Gauge;
-import io.prometheus.client.Histogram;
+import io.camunda.zeebe.util.collection.Map3D;
+import io.camunda.zeebe.util.collection.Table;
+import io.camunda.zeebe.util.micrometer.MicrometerUtil;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.util.HashMap;
+import java.util.Map;
 
 final class MessagingMetricsImpl implements MessagingMetrics {
 
-  private static final String NAMESPACE = "zeebe";
-  private static final String LABEL_TOPIC = "topic";
-  private static final String LABEL_ADDRESS = "address";
-  private static final String REQ_TYPE_MESSAGE = "MESSAGE";
-  private static final String REQ_TYPE_REQ_RESP = "REQ_RESP";
-
-  private static final Histogram REQUEST_RESPONSE_LATENCY =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("messaging_request_response_latency")
-          .help("The time how long it takes to retrieve a response for a request")
-          .labelNames(LABEL_TOPIC)
-          .register();
-
-  private static final Histogram REQUEST_SIZE_IN_KB =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("messaging_request_size_kb")
-          .help("The size of the request, which has been sent")
-          .labelNames(LABEL_ADDRESS, LABEL_TOPIC)
-          .buckets(.01, .1, .250, 1, 10, 100, 500, 1_000, 2_000, 4_000)
-          .register();
-  private static final Counter REQUEST_COUNT =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("messaging_request_count")
-          .help("Number of requests which has been send to a certain address")
-          .labelNames("type", LABEL_ADDRESS, LABEL_TOPIC)
-          .register();
-
-  private static final Counter RESPONSE_COUNT =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("messaging_response_count")
-          .help("Number of responses which has been received")
-          .labelNames(LABEL_ADDRESS, LABEL_TOPIC, "outcome")
-          .register();
-
-  private static final Gauge IN_FLIGHT_REQUESTS =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("messaging_inflight_requests")
-          .help("The count of inflight requests")
-          .labelNames(LABEL_ADDRESS, LABEL_TOPIC)
-          .register();
+  private static final String MESSAGE = "MESSAGE";
+  private static final String REQ_RESP = "REQ_RESP";
```

Comment:
:wrench: If I understood correctly, these are the two possible values for the type label? We should make it an enum.